core/unit_test/TestSharedAlloc.hpp

```diff
@@ -82,7 +82,8 @@ void test_shared_alloc() {
     // Sanity check for the whole set of allocation records to which this record
     // belongs.
     RecordBase::is_sane(r[0]);
-    // RecordMemS::print_records( std::cout, s, true );
+    [[maybe_unused]] std::stringstream dummy;
```

Comment:
The `[[maybe_unused]]` attribute is unnecessary.
I would prefer to revert these changes.  They don't belong to a PR that claims to be a hotfix for the CI